src/index.ts

```diff
@@ -79,7 +79,7 @@ export const initialize = ({
       [SemanticResourceAttributes.SERVICE_NAME]:
         serviceName ?? UNKNOWN_SERVICE_NAME,
     }),
-    sampler: new TraceIdRatioBasedSampler(samplingProbability),
+    sampler: new TraceIdRatioBasedSampler(tryNumber(samplingProbability)),
```

Comment:
`samplingProbability` is also later used as the exporter attribute. We should convert it to number there as well, because OTLP attributes have types.